simplesat/sat/policy/undetermined_clause_policy.py

```diff
@@ -16,17 +15,23 @@ class UndeterminedClausePolicy(IPolicy):
     truth value is not yet known and suggests them in descending order by
     package version number. """
 
-    def __init__(self, pool, installed_repository, prefer_installed=True):
+    def __init__(self, pool, installed_repository,
+                 ignore_installed_packages=None):
+        if ignore_installed_packages is None:
+            ignore_installed_packages = set()
         self._pool = pool
-        self.prefer_installed = prefer_installed
+
         by_version = six.functools.partial(pkg_id_to_version, self._pool)
-        self._installed_ids = sorted(
-            (pool.package_id(package) for package in installed_repository),
-            key=by_version
-        )
-        self._preferred_package_ids = {
+        installed_packages = set(package for package in installed_repository)
+        prefer_installed = installed_packages - ignore_installed_packages
```

Comment:
we can rename `prefer_installed` to `preferred_installed_packages` ? Seems more consistent